lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,356 @@
+             $,
+             _,
+             gettext,
+             moment,
+             HtmlUtils,
+             EntitlementModel,
+             CourseCardModel,
+             pageTpl,
+             verificationPopoverTpl
+         ) {
+             return Backbone.View.extend({
+                 tpl: HtmlUtils.template(pageTpl),
+                 verificationTpl: HtmlUtils.template(verificationPopoverTpl),
+
+                 events: {
+                     'change .session-select': 'updateEnrollBtn',
+                     'click .enroll-btn': 'handleEnrollChange',
+                     'click .popover-dismiss': 'hideVerificationDialog',
+                     'keydown .final-confirmation-btn': 'handleVerificationPopoverA11y'
+                 },
+
+                 initialize: function(options) {
+                     this.$el = $(this.el);
+
+                     // Set up models and reload view on change
+                     this.courseCardModel = new CourseCardModel();
+                     this.entitlementModel = new EntitlementModel({
+                         availableSessions: this.formatDates(JSON.parse(options.availableSessions)),
+                         entitlementUUID: options.entitlementUUID,
+                         currentSessionId: options.currentSessionId,
+                         userId: options.userId
+                     });
+                     this.listenTo(this.entitlementModel, 'change', this.render);
+
+                     // Grab URLs that handle changing of enrollment and entering a newly enrolled session.
+                     this.enrollUrl = options.enrollUrl;
+                     this.courseHomeUrl = options.courseHomeUrl;
+
+                     // Grab elements from the parent card that work with this view and bind associated events
+                     this.$triggerOpenBtn = $(options.triggerOpenBtn) || $(''); // Opens/closes session selection view
+                     this.$dateDisplayField = $(options.dateDisplayField) || $(''); // Displays current session dates
+                     this.$enterCourseBtn = $(options.enterCourseBtn) || $(''); // Button link to course home page
+                     this.$courseCardMessages = $(options.courseCardMessages) || $(''); // Additional session messages
+                     this.$courseTitleLink = $(options.courseTitleLink) || $(''); // Title link to course home page
+                     this.$courseImageLink = $(options.courseImageLink) || $(''); // Image link to course home page
+                     this.$triggerOpenBtn.on('click', this.toggleSessionSelectionPanel.bind(this));
+
+                     this.render(options);
+                     this.postRender();
+                 },
+
+                 render: function() {
+                     HtmlUtils.setHtml(this.$el, this.tpl(this.entitlementModel.toJSON()));
+                     this.delegateEvents();
+                     this.updateEnrollBtn();
+                     return this;
```

Comment:
Is this needed for tests? If not I don't think you should need to include a return statement.

Reply:
Yes, exactly, the tests require it so that they can check the returned object.

Reply:
Cool